WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -673,19 +673,15 @@ def stripe_token_id(options = {})
       let(:competition) { FactoryBot.create(:competition, :entry_fee, :visible, :registration_open, organizers: [organizer], events: Event.where(id: %w(222 333))) }
       let!(:registration) { FactoryBot.create(:registration, competition: competition, user: organizer) }
 
-      it 'processes a payment then issues a refund' do
+      before :each do
         sign_in organizer
-        token_id = Stripe::Token.create(
-          card: {
-            number: "4242424242424242",
-            exp_month: 12,
-            exp_year: Date.today.year + 1,
-            cvc: "314",
-          },
-        ).id
+        token_id = stripe_token_id
         post :process_payment, params: { competition_id: competition.id, payment: { stripe_token: token_id, total_amount: registration.outstanding_entry_fees.cents } }
+      end
+
+      it 'processes a payment then issues a full refund' do
```

Comment:
Some more testing nit picks for you, but this LGTM and feel free to merge it whenever you're happy with it!

All of these `it`s still have some shared stuff: they all start with the line `payment = ...` and they all have "processes a payment" in their description:

1. To share the payment variable throughout the specs, you can add a `@payment = ...` to the end of the `before :each ... end` block, and then reference `@payment` in the specs.
2. You could also restructure the tests a bit inside of a `context` block:

```ruby
context "processes a payment" do
  before: each do ... end

  it 'issues a full refund' do ... end
  it 'issues a 50% refund' do ... end
  it 'disallows negative refund' do ... end
  it 'disallows a refund more than the payment' do ... end
end
```

Reply:
That does look better. I'll make those suggested changes.